src/simply/simply_window_stack.c

```diff
@@ -31,31 +31,57 @@ struct __attribute__((__packed__)) WindowShowPacket {
 
 typedef struct WindowSignalPacket WindowSignalPacket;
 
-struct __attribute__((__packed__)) WindowSignalPacket {
-  Packet packet;
-  uint32_t id;
-};
+#define WindowSignalPacketDef { \
+  Packet packet;                \
+  uint32_t id;                  \
+}
+
+struct __attribute__((__packed__)) WindowSignalPacket WindowSignalPacketDef;
 
 typedef WindowSignalPacket WindowHidePacket;
 
 typedef WindowHidePacket WindowEventPacket;
 
+struct __attribute__((__packed__)) WindowLoadEventPacket {
+  union {
+    struct WindowSignalPacketDef;
+  };
```

Comment:
I think I enabled `-ms-extensions` in the wscript, so you can just have 

```
struct __attribute__((__packed__)) WindowLoadEventPacket {
  WindowSignalPacket;
  GSize size;
};
```

Let me know how this goes. I haven't used that extensions across the entire codebase yet. I also sometimes wonder if I should have enabled it, but it probably beats using a macro.